labgrid/driver/sigrokdriver.py

```diff
@@ -329,3 +329,91 @@ def measure(self):
         if len(res) != 2:
             raise ExecutionError(f"Cannot parse --show output {out}")
         return res
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class SigrokDmmDriver(SigrokCommon):
+    """
+    This driver wraps around a single channel DMM controlled by sigrok.
+    It has been tested with Unit-T UT61C and UT61B devices but probably also
+    works with other single chnnel DMMs.
+
+    This driver binds to a SigrokUsbDevice.
+    Make sure to select the correct driver for your DMM there.
+
+    Example usage:
+    > resources:
+    >   - SigrokUSBDevice:
+    >       driver: uni-t-ut61c
+    >       match:
+    >         'ID_PATH': pci-0000:07:00.4-usb-0:2:1.0
+    > drivers:
+    >   - SigrokUt61DmmDriver: {}
+
+    Args:
+        bindings (dict): driver to use with sigrok
+    """
+    bindings = {
+        "sigrok": {SigrokUSBSerialDevice, NetworkSigrokUSBSerialDevice, SigrokUSBDevice, NetworkSigrokUSBDevice},
+    }
+
+    @Driver.check_active
+    @step(result=True)
+    def start(self, samples, timeout=None):
+        """
+        Starts to read samples from the DMM.
+        This function returns once sampling has been started. Sampling continues in the background.
+
+        Note: We use subprocess.PIPE to buffer the samples. When this buffer is too small for the number of
+        sampled requested sampling may stall.
+
+        Args:
+            samples: Number of samples to obtain
+            timeout: Timeout after which sampling should be stopped.
+                     If None: timemout[s] = samples * 1s + 5s
+                     If int: Timeout in [s]
+
+        Returns: None
+        """
+        if not timeout:
+            timeout = samples + 5
+
+        args = f"-O csv --samples {samples}".split(" ")
+        self._call_with_driver(*args)
+        self._timeout = time.time() + timeout
+
+    @Driver.check_active
+    @step(result=True)
+    def get_samples(self):
+        """
+        Waits for sigrok to complete and returns all samples obtained afterwards.
+        This function blocks until either sigrok has terminated or the timeout has been reached.
+
+        Returns:
+            (unit_spec, [sample, ...])
+        """
+        assert self._timeout
```

Comment:
Hm, `self._timeout` is only defined when `start()` was called before, so that won't even raise an AssertionError, but an AttributeError. I think that's bad style. Maybe set SigrokCommon's `_running` in `start()` and check for that?

Reply:
Totally right. I am now using `_running` - that's better style anyway. Thanks for the pointer.